util/mmio.h

```diff
@@ -69,10 +69,64 @@
 #ifdef __s390x__
 #include <unistd.h>
 #include <sys/syscall.h>
-
-/* s390 requires a privileged instruction to access IO memory, these syscalls
-   perform that instruction using a memory buffer copy semantic.
+#include <sys/auxv.h>
+
+/* s390 requires special instructions to access IO memory. Originally there
```

Comment:
This stuff should not be compiled except on s390, and probably be in its own header too